hil-test/src/bin/rmt.rs

```diff
@@ -333,11 +333,15 @@ fn do_rmt_single_shot<const TX_LEN: usize>(
         .with_clk_divider(DIV)
         .with_memsize(tx_memsize);
 
-    let (tx_channel, _) = ctx.setup_loopback(tx_config, Default::default());
+    let (tx_channel, _) = ctx.setup_loopback(&tx_config, &Default::default());
 
     let tx_data: [_; TX_LEN] = generate_tx_data(write_end_marker, write_end_marker);
 
-    tx_channel.transmit(&tx_data)?.wait().map_err(|(e, _)| e)?;
+    tx_channel
+        .transmit(&tx_data)
+        .map_err(|(e, _)| e)?
```

Comment:
Have you considered just unwrapping here? That actually gives us a backtrace closer to where the error originated from.

Reply:
The `rmt_single_shot_fails_without_end_marker` inspects the exact error type.

That said; I have another PR in the pipeline that refactors RMT HIL tests to clean them up a bit, and I plan to remove the other single-shot tests entirely: They don't add value over the loopback tests (except for `rmt_single_shot_fails_without_end_marker`). I added them a while ago (in https://github.com/esp-rs/esp-hal/pull/3477) when stuck debugging loopback tests.